stake-pool/program/src/instruction.rs

```diff
@@ -418,24 +421,27 @@ pub fn update_stake_pool_balance(
     })
 }
 
-/// Creates a 'Deposit' instruction.
+/// Creates instructions required to deposit into a stake pool, given a stake
+/// account owned by the user
 pub fn deposit(
     program_id: &Pubkey,
     stake_pool: &Pubkey,
     validator_list_storage: &Pubkey,
-    stake_pool_deposit: &Pubkey,
-    stake_pool_withdraw: &Pubkey,
+    stake_pool_deposit_authority: &Pubkey,
+    stake_pool_withdraw_authority: &Pubkey,
     stake_to_join: &Pubkey,
     validator_stake_accont: &Pubkey,
     pool_tokens_to: &Pubkey,
     pool_mint: &Pubkey,
     token_program_id: &Pubkey,
-) -> Result<Instruction, ProgramError> {
+    current_stake_authority: &Pubkey,
```

Comment:
nit: let's place `current_stake_authority` next to `stake_to_join` in the argument list, perhaps:
```
...
deposit_stake_address: &Pubkey,
deposit_stake_withdraw_authority: &Pubkey,
...
```